### PR TITLE
[Jest] Migrate `marker.test.js`

### PR DESCRIPTION
**Pull request by wipfli**
- Move and rename
- Run jscodeshift
- Run custom codemods and lint
- Manual edits

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Manually test the debug page.
 - [ ] Suggest a changelog category: bug/feature/docs/etc. or "skip changelog".
 - [ ] Add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`.

### CODE REVIEW

**Comment by wipfli** [comment]
Works if the argument of the popup constructor is optional. This change is in my last pull request.﻿

**Comment by wipfli** [comment]
@HarelM this one is ready now as well.

**Comment by wipfli** [comment]
View diff of manual edits with https://github.com/maplibre/maplibre-gl-js/compare/bd1b71a945d5bc791e88edbe21534a6be73440d4..45b0e1eca4c1149c6c0f5624bb7adbef5f661c92

File: src/ui/marker.test.ts [hunk 1]
```diff
@@ -0,0 +1,774 @@
...
+
+    test('Popup offsets around default Marker', () => {
+        const map = createMap();
+
+        const marker = new Marker()
+            .setLngLat([0, 0])
+            .setPopup(new Popup().setText('Test'))
+            .addTo(map);
+
+        expect(marker.getPopup().options.offset['bottom'][1] < 0).toBeTruthy();
```
Comment (HarelM) [inline src/ui/marker.test.ts:254]: Why did you change this? Is the return type from getPopup().options incorrect?

Comment (wipfli) [reply src/ui/marker.test.ts:254]: Because of the following error:
```
    src/ui/marker.test.ts:254:49 - error TS2339: Property 'bottom' does not exist on type 'Offset'.
      Property 'bottom' does not exist on type 'number'.

    254         expect(marker.getPopup().options.offset.bottom[1] < 0).toBeTruthy();
```

Comment (wipfli) [reply src/ui/marker.test.ts:254]: Which is a bit strange given

image

Comment (wipfli) [reply src/ui/marker.test.ts:254]: https://stackoverflow.com/questions/38324949/error-ts2339-property-x-does-not-exist-on-type-y

Comment (HarelM) [reply src/ui/marker.test.ts:254]: It's not strange since the compiler doesn't know which type you assume this is so it is checking all the given types and not all of them fix.
In order to over come it the best approach would be to type the following type in popup.ts file:
i.e. instead of using:
```ts
export type Offset = number | PointLike | {
  [_ in PositionAnchor]: PointLike;
};
```
Write something like 
```ts
export type PositionPointLike = [_ in PositionAnchor]: PointLike; // find a better name
export type Offset = number | PointLike | PositionPointLike;
```

And in your test do
```ts
let offset = marker.getPopup().options.offset as PositionPointLike;
...
```

Comment (wipfli) [reply src/ui/marker.test.ts:254]: Actually I would like to revoke my thumbs up. The way it is done now is already the best solution. One option to do it closer to the original way would be:

```js

        const offset = marker.getPopup().options.offset as Record<PositionAnchor, PointLike>;

        expect(offset.bottom[1] < 0).toBeTruthy();
        expect(offset.top[1] === 0).toBeTruthy();
        expect(offset.left[0] > 0).toBeTruthy();
        expect(offset.right[0] < 0).toBeTruthy();

        expect(marker.getPopup().options.offset['bottom-left'][0] > 0).toBeTruthy();
        expect(marker.getPopup().options.offset['bottom-left'][1] < 0).toBeTruthy();
        expect(marker.getPopup().options.offset['bottom-right'][0] < 0).toBeTruthy();
        expect(marker.getPopup().options.offset['bottom-right'][1] < 0).toBeTruthy();

        expect(marker.getPopup().options.offset['top-left']).toEqual([0, 0]);
        expect(marker.getPopup().options.offset['top-right']).toEqual([0, 0]);
```

But that only works for the `PositionAnchor`s that don't have a dash. How important is this to you @HarelM ?

Comment (HarelM) [reply src/ui/marker.test.ts:254]: Not that important, which ever you find is more readable.

Comment (wipfli) [reply src/ui/marker.test.ts:254]: Great. Thanks. I would like to leave it as it is now. Thanks for reviewing all the other pull requests by the way :)

**Review by HarelM — APPROVED** [review]
